implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -244,26 +262,39 @@ Metric writeHistogramSupport(HistogramSupport histogramSupport) {
                 histogramDataPoint.addBucketCounts((long) countAtBucket.count());
             }
             metricBuilder.setHistogram(Histogram.newBuilder()
-                    .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                    .setAggregationTemporality(AggregationTemporality.forNumber(config.getAggregationTemporality()))
                     .addDataPoints(histogramDataPoint));
             return metricBuilder.build();
         }
 
         return metricBuilder.setHistogram(Histogram.newBuilder()
-                .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                .setAggregationTemporality(AggregationTemporality.forNumber(config.getAggregationTemporality()))
                 .addDataPoints(histogramDataPoint)).build();
     }
 
     // VisibleForTesting
     Metric writeFunctionTimer(FunctionTimer functionTimer) {
         return getMetricBuilder(functionTimer.getId()).setHistogram(Histogram.newBuilder()
                 .addDataPoints(HistogramDataPoint.newBuilder().addAllAttributes(getTagsForId(functionTimer.getId()))
-                        .setStartTimeUnixNano(((StartTimeAwareMeter) functionTimer).getStartTimeNanos())
-                        .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                        .setStartTimeUnixNano(getStartTimeNanos((functionTimer))).setTimeUnixNano(getEndTimeNanos())
                         .setSum(functionTimer.totalTime(getBaseTimeUnit())).setCount((long) functionTimer.count())))
                 .build();
     }
 
+    private boolean isDeltaAggregationTemporality() {
+        return config.getAggregationTemporality() == AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA_VALUE;
+    }
+
+    private long getStartTimeNanos(Meter meter) {
+        return isDeltaAggregationTemporality() ? getEndTimeNanos() - config.step().toNanos()
+                : ((StartTimeAwareMeter) meter).getStartTimeNanos();
+    }
+
+    private long getEndTimeNanos() {
+        return isDeltaAggregationTemporality() ? (clock.wallTime() / config.step().toMillis()) * config.step().toNanos()
```

Comment:
Going to work more on this and the tests too. Just wanted to kick start this PR with an initiative for Delta-based Aggregation Temporality.
But the overall idea is not to use different times for Step times. What I tried to achieve is to have normalized steps like this - 00:00:01.000 - 00:00:02.00, 00:00:02.000 - 00:00:03.00. This will make sure all the exported meters for a given step will have consistent timing.

Reply:
Tried to fix this in the lastest commit.